homeassistant/components/sensor/nzbget.py

```diff
@@ -0,0 +1,146 @@
+    "DownloadPaused": ("Download Paused", None),
+    "FreeDiskSpaceMB": ("Disk Free", "MB"),
+    "PostPaused": ("Post Processing Paused", None),
+    "RemainingSizeMB": ("Queue Size", "MB"),
+}
+DEFAULT_TYPES = [
+    "DownloadRate",
+    "DownloadPaused",
+    "RemainingSizeMB",
+]
+
+_LOGGER = logging.getLogger(__name__)
+_THROTTLED_REFRESH = None
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Sets up nzbget sensors"""
+
+    base_url = config.get("base_url")
+    name = config.get("name", "NZBGet")
+    username = config.get("username")
+    password = config.get("password")
+    monitored_types = config.get("monitored_variables", DEFAULT_TYPES)
+
+    if not base_url:
+        _LOGGER.error("Missing base_url config for NzbGet")
+        return False
+
+    url = "{}/jsonrpc".format(base_url)
+
+    nzbgetapi = NZBGetAPI(api_url=url, username=username, password=password)
+    global _THROTTLED_REFRESH
+    _THROTTLED_REFRESH = Throttle(timedelta(seconds=1))(nzbgetapi.update)
+
+    devices = []
+    for ng_type in monitored_types:
+        new_sensor = NZBGetSensor(api=nzbgetapi,
+                                  sensor_type=ng_type,
+                                  client_name=name)
+        devices.append(new_sensor)
+    add_devices(devices)
+
+
+class NZBGetAPI(object):
+    """Simple json-rpc wrapper for nzbget's api"""
+    def __init__(self, api_url, username=None, password=None):
+        self.api_url = api_url
+        self.status = None
+        self.last_update = None
+        self.headers = {'content-type': 'application/json'}
+        if username is not None and password is not None:
+            self.auth = (username, password)
+        else:
+            self.auth = None
+
+    def post(self, method, params=None):
+        """Sends a post request, and returns the response as a dict."""
+        payload = {"method": method}
+        if params:
+            payload['params'] = params
+        try:
+            response = requests.post(self.api_url,
+                                     json=payload,
+                                     auth=self.auth,
+                                     headers=self.headers)
+            return response.json()
+        except requests.exceptions.ConnectionError as e:
+            _LOGGER.error("Failed to update nzbget status from %s.  Error: %s",
+                          self.api_url, e)
+            return False
+
+    def update(self):
+        """Updates cached response"""
+        try:
+            self.status = self.post('status')['result']
+            self.last_update = int(time.time())
+        except TypeError:
+            # failed to update status
+            return False
+
+
+class NZBGetSensor(Entity):
+    """Represents an NZBGet sensor"""
+    def __init__(self, api, sensor_type, client_name):
+        try:
+            self._name = SENSOR_TYPES[sensor_type][0]
+        except KeyError:
```

Comment:
Returning False does nothing here. You should verify this before instantiating this class.